### PR TITLE
random string if nickname already used

### PR DESCRIPTION
**Pull request by RobinBurri**
When the user is created in the DB with the nickname of 42, it checks that it is not already in use and if it is, it create a random 5 char string as a nickname.

### CODE REVIEW

**Review by shaolin-peanut — APPROVED** [review]

File: backend/src/auth/auth.service.ts [hunk 1]
```diff
@@ -11,16 +11,33 @@ import { UserService } from 'src/user/user.service'
...
     constructor(private userService: UserService) {}
 
     async validateUser(user42) {
         let user = await this.userService.findByFT_id(user42.FT_id)
         if (!user) {
+            let newNickname = user42.nickname
+            const usersameNickname = await this.userService.findByNickname(
+                user42.nickname
+            )
+            if (usersameNickname) newNickname = generateRandomString()
```
Comment (t-h2o) [inline backend/src/auth/auth.service.ts:37]: Use Camel Case
```suggestion
            const userSameNickname = await this.userService.findByNickname(
                user42.nickname
            )
            if (userSameNickname) newNickname = generateRandomString()
```

File: backend/src/auth/auth.service.ts [hunk 2]
```diff
@@ -11,16 +11,33 @@ import { UserService } from 'src/user/user.service'
...
     constructor(private userService: UserService) {}
 
     async validateUser(user42) {
         let user = await this.userService.findByFT_id(user42.FT_id)
         if (!user) {
+            let newNickname = user42.nickname
+            const userSameNickname = await this.userService.findByNickname(
+                user42.nickname
+            )
+            if (userSameNickname) newNickname = generateRandomString()
```
Comment (t-h2o) [inline backend/src/auth/auth.service.ts:37]: Should we do a loop ?
```js
for (userSameNickname = await this.userService.findByNickname(user42.nickname); userSameNickname; userSameNickname = await this.userService.findByNickname(newNickname))
{
  newNickname = generateRandomString();
}
```
In this way, we are sure to not have double name...

Comment (RobinBurri) [reply backend/src/auth/auth.service.ts:37]: How can there be same name multiple times, already present in the db?

Comment (t-h2o) [reply backend/src/auth/auth.service.ts:37]: I don' t know, perhaps if we scale our application for millions users...

File: backend/src/auth/auth.service.ts [hunk 3]
```diff
@@ -11,16 +11,33 @@ import { UserService } from 'src/user/user.service'
...
             user = await this.userService.create({
                 FT_id: user42.FT_id,
-                nickname: user42.nickname,
+                nickname: newNickname,
```
Comment (t-h2o) [inline backend/src/auth/auth.service.ts:40]: In this way we have the login nickname somewhere in the nickname
```suggestion
                nickname: `${user42.nickname}-${randomString},
```

Comment (RobinBurri) [reply backend/src/auth/auth.service.ts:40]: Casse pas les couilles Theo

File: backend/src/auth/auth.service.ts (hunk 2 above)
Comment (RobinBurri) [reply backend/src/auth/auth.service.ts:37]: How can there be same name multiple times, already present in the db?
Explain to me

**Review by t-h2o — APPROVED** [review]

File: backend/src/auth/auth.service.ts (hunk 2 above)
Comment (t-h2o) [reply backend/src/auth/auth.service.ts:37]: if this function `Math.random()` return the same sequence of numbers

Comment (t-h2o) [reply backend/src/auth/auth.service.ts:37]: You need to create a random string twice, and the string has already been created.

**Pull request merged**